core/components/handler/pktStorage.go

```diff
@@ -145,6 +147,7 @@ func (e pktEntry) MarshalBinary() ([]byte, error) {
 	rw := readwriter.New(nil)
 	rw.Write(e.AppEUI)
 	rw.Write(e.DevEUI)
+	rw.Write(uint16(e.FPort))
```

Comment:
I vote for option 2 as well as we're currently not able to send any downlink data anyway... So deploying it now would not make a difference.